Lib9c/Action/RewardGold.cs

```diff
@@ -169,18 +173,28 @@ public IAccountStateDelta PrepareNextArena(IActionContext ctx, IAccountStateDelt
                                 }
                             }
 
-                            // Copy ArenaInfo from prev ArenaInfo.
-                            foreach (var address in addressList)
+                            // Copy list for loop.
+                            var enumerator = addressList.ToList();
+
+                            // Copy activated ArenaInfo from prev ArenaInfo.
+                            foreach (var address in enumerator)
                             {
                                 if (states.TryGetState(
                                         prevWeekly.address.Derive(address.ToByteArray()),
                                         out Dictionary rawInfo))
                                 {
                                     var prevInfo = new ArenaInfo(rawInfo);
-                                    var info = new ArenaInfo(prevInfo);
+                                    var record = prevInfo.ArenaRecord;
+                                    // Filter ArenaInfo
+                                    if (filterInactive && record.Win == 0 && record.Draw == 0 &&
+                                        record.Lose == 0)
+                                    {
+                                        addressList.Remove(address);
```

Comment:
Instead of maintaining `addressList` itself, how about using new aggregated value for `listAddress` state?